plugins/jobtype/src/azkaban/jobtype/tuning/TuningParameterUtils.java

```diff
@@ -0,0 +1,214 @@
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.apache.log4j.Logger;
+
+import azkaban.jobtype.HadoopConfigurationInjector;
+import azkaban.utils.Props;
+
+import com.google.gson.Gson;
+
+/**
+ * TuningParameterUtils is a utility class responsible for updating tuned hadoop parameters for the job
+ * by calling tuning api. Before calling tuning api, it gets default parameters for the job and passes default
+ * parameters to api. By default it tries 3 times in case api is down before running the job with default parameters.
+
+ */
+public class TuningParameterUtils {
+
+  private static Logger log = Logger.getRootLogger();
+  private static final int DEFAULT_TUNING_API_RETRY_COUNT = 3;
+  private static final int DEFAULT_TUNING_API_TIMEOUT_MS = 10000;
+
+  private static final String USER_TO_PROXY = "user.to.proxy";
+  private static final String TUNING_API_RETRY_COUNT = "tuning.api.retry,count";
+  private static final String TUNING_API_TIMEOUT = "tuning.api.timeout";
+  private static final String AZKABAN = "azkaban";
+  private static final String TUNING_API_VERSION = "2";
+  private TuningParameterUtils() {
+
+  }
+
+  /**
+   * Get hadoop properties from the job
+   * @param props
+   * @return Map of parameter name and value
+   */
+  public static Map<String, String> getHadoopProperties(Props props) {
+    Map<String, String> confProperties = props.getMapByPrefix(HadoopConfigurationInjector.INJECT_PREFIX);
+    if (log.isDebugEnabled()) {
+      for (Map.Entry<String, String> entry : confProperties.entrySet()) {
+        log.debug("Adding default parameter " + entry.getKey() + ", value " + entry.getValue());
+      }
+    }
+    return confProperties;
+  }
+
+  /**
+   * This method calls Auto Tuning Framework Rest API and adds new parameters in props.
+   * This method will retry 3 times in case there is timeout while calling rest API.
+   * @param props
+   */
+  public static void updateAutoTuningParameters(Props props) {
+    int count = 0;
+    boolean noException = false;
+    Map<String, String> params = null;
+    if(props.containsKey(TuningCommonConstants.TUNING_API_END_POINT)==false)
+    {
+      log.error("Tuning API End Point not found in properties. Not applying auto tuning. ");
+      return;
+    }
+    int retryCount = DEFAULT_TUNING_API_RETRY_COUNT;
+    if (props.containsKey(TUNING_API_RETRY_COUNT)) {
+      retryCount = props.getInt(TUNING_API_RETRY_COUNT);
+    }
+
+    while (!noException && count < retryCount) {
+      try {
+        count++;
+        log.info("Calling get current run parameters. Try count " + count);
+        params = getCurrentRunParameters(props);
+        log.info("Parameters are " + params.toString());
+        noException = true;
+      } catch (IOException e) {
+        log.error("Error in getting current run parameter ", e);
+      } catch (Exception e) {
```

Comment:
Should we really catch all Exception here? If so, then these two catch blocks should be consolidated

Reply:
Consolidated into one.